### PR TITLE
Update the project documentation + migration guide [ECR-3013]:

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview
Add the migration guide and update the project documentation.

---
See: https://jira.bf.local/browse/ECR-3013

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+To update an existing service:
+
+1. Configure a maven-assembly-plugin or maven-shade-plugin to create
+   a single JAR with all service dependencies.
+   See [How to Build Service Artifact][how-to-build-docs] section for details.
+1. Add the following manifest entries to the JAR:
+    - "Plugin-Id" equal to "${project.groupId}:${project.artifactId}:${project.version}"
+    - "Plugin-Version" equal to "${project.version}"
+1. Add a "provided"-scoped dependency on PF4J.
```
Comment (dmitry-timofeev) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: I wonder if the annotation processor present in PF4J will kick in if it remains a transitive dependency of Exonum? In other words, do users **have to** specify it explicitly?

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+</details>
+
+## Update the node configuration
+
+Below are the highlights of the changes in node configuration:
+
+1. It is no longer needed to pass the system nor service classpath.
+   Paths to service artifacts are specified in "ejb_app_services.toml" 
+   configuration file in the *working directory* of each node application.
+1. Some configuration options were moved between configuration phases.
```
Comment (dmitry-timofeev) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: I think it is not worth documenting each changed parameter, as they are listed in the tutorial for each phase. Is a single link to it enough?

File: exonum-java-binding/CHANGELOG.md [hunk 3]
```diff
@@ -15,8 +15,13 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+**If you are upgrading an existing Java service, consult 
+the [migration guide](https://github.com/exonum/exonum-java-binding/blob/ejb/v0.6.0/exonum-java-binding/doc/Migration_guide_0.6.md).**
+
 ## [0.5.0] - 2019-03-13
 
+- Support multiple simultaneously active Java services on the network.
```
Comment (vitvakatu) [inline exonum-java-binding/CHANGELOG.md]: Why for 0.5?

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+### Service Packaging Requirements
+ 
+Exonum Java 0.6.0 brings a new packaging format of services. Service
+classes and resources must be packaged in a single JAR archive
+with some metadata set.
+
+To update an existing service:
+
+1. Configure a maven-assembly-plugin or maven-shade-plugin to create
```
Comment (vitvakatu) [inline exonum-java-binding/doc/Migration_guide_0.6.md]:
```suggestion
1. Configure a `maven-assembly-plugin` or `maven-shade-plugin` to create
```

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+Below is a diff of changes applied to the cryptocurrency-demo service
+to support the new service artifact format. Your service implementation
+is likely to require similar changes:
+
+<details>
+<summary>
+Changes to cryptocurrency service since 0.5.0
+</summary>
+
+<!-- This diff is not the easiest thing to read, maybe, just give
```
Comment (vitvakatu) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: Indeed, not really readable

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+a git diff command (the output will, however, include some unrelated
+changes? --> 
+```
+Index: exonum-java-binding/cryptocurrency-demo/pom.xml
```
Comment (bullet-tooth) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: O believe `git diff` is a better idea then having such struct output here

File: exonum-java-binding/doc/Migration_guide_0.6.md (hunk 1 above)
Comment (Maria-Nosyk) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: Add a dependency on PF4J within the provided scope.

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
++public final class CryptocurrencyServiceModule extends AbstractServiceModule {
+ 
+   @Override
+   protected void configure() 
+
+```
+
+</details>
+
+## Update the node configuration
```
Comment (Maria-Nosyk) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: Update the Node Configuration

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+Below are the highlights of the changes in node configuration:
```
Comment (Maria-Nosyk) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: in the node configuration

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+1. It is no longer needed to pass the system nor service classpath.
```
Comment (Maria-Nosyk) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: It is no longer required to pass either the system or the service classpath.

File: exonum-java-binding/doc/Migration_guide_0.6.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+   Paths to service artifacts are specified in "ejb_app_services.toml" 
+   configuration file in the *working directory* of each node application.
```
Comment (Maria-Nosyk) [inline exonum-java-binding/doc/Migration_guide_0.6.md]: in the "ejb_app_services.toml" configuration file

**Comment by dmitry-timofeev** [comment]
Merged to be able to cut the release branch — any corrections/suggestions will be addressed in `ejb/v0.6.0-release`.

**Pull request merged**